module/VuFind/src/VuFind/Controller/AbstractRecord.php

```diff
@@ -549,7 +549,10 @@ public function emailAction()
                 $this->flashMessenger()->addMessage('email_success', 'success');
                 return $this->redirectToRecord();
             } catch (MailException $e) {
-                $this->flashMessenger()->addMessage($e->getMessage(), 'error');
+                $this->flashMessenger()->addMessage(
+                    $e->getCode() ? $e->getMessage() : 'email_failure',
```

Comment:
Since this logic is repeated a number of times, do you think it would be better to put it inside a custom method in `VuFind\Exception\Mail`? That would also open up the possibility of implementing different behavior when we're in development mode (i.e. only suppress the technical message when we're _not_ in dev mode). One approach would be to override the `getMessage` method, but the results of doing that might be less predictable than creating a new method like `getSafeUserMessage` or `getDisplayMessage`. Since the code is always specifically catching a `MailException`, it should be safe to rely on the existence of such a method.

Reply:
I'd also suggest that maybe it's better to explicitly say `$e->getCode() !== ERROR_DEFAULT` rather than relying on the non-obvious fact that `ERROR_DEFAULT` has a falsy value. And maybe `ERROR_DEFAULT` should be `ERROR_UNKNOWN` or `ERROR_TECHNICAL` or something that better reflects its use here.

Reply:
Good points, I'll alter the checks.